packages/tooltip/src/Tooltip.tsx

```diff
@@ -232,8 +234,8 @@ function Tooltip({
   const tooltipRef = useRef<HTMLDivElement>(null);
 
   const tooltipId = useMemo(
-    () => id || `tooltip-${Math.floor(Math.random() * Math.floor(10))}`,
-    [id],
+    () => id ?? tooltipRef.current?.id ?? idAllocator.generate(),
+    [id ?? tooltipRef.current?.id],
```

Comment:
1. Are you able to change the values of the dependency array? TIL :woah:
2. Won't these be the same? If `id` is not set then wouldn't `tooltipRef.current?.id` be unset as well?

Reply:
>Are you able to change the values of the dependency array? TIL :woah:

could you clarify what you mean by "change the values"? 

> Won't these be the same? If id is not set then wouldn't tooltipRef.current?.id be unset as well?

On initial render `tooltipRef.current?.id` won't be set, but on subsequent renders it will be `id` if it's set, or a generated ID

Reply:
```
[id ?? tooltipRef.current?.id]
```

If the `id` prop changes then the dependency array for the useMemo will change - that's allowed?

Reply:
> If the `id` prop changes then the dependency array for the useMemo will change - that's allowed?

it's not really any different from assigning to an intermediate variable, i.e.
```tsx
const givenId = id ?? tooltipRef.current?.id;
useMemo(
  ...,
  [givenId],
);
```

React doesn't actually know or care what goes into a dependency array; it's effectively just an array whose shallow comparison is used to determine whether an operation should be recomputed.

Reply:
Yeah I don't see it mentioned anywhere on the docs, but I know Hooks has weird rules.